src/account_manager.rs

```diff
@@ -82,6 +82,51 @@ impl AccountManager {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    #[tracing::instrument(skip(self, protocol_store))]
+    pub async fn check_pre_keys<P: PreKeysStore>(
+        &mut self,
+        protocol_store: &mut P,
+        service_id_kind: ServiceIdKind,
+    ) -> Result<bool, ServiceError> {
+        let Some(signed_prekey_id) = protocol_store.signed_prekey_id().await?
+        else {
+            tracing::warn!("No signed prekey found");
+            return Ok(false);
+        };
+        // XXX: should we instead use the `load_last_resort_kyber_pre_keys` method? Or refactor
+        //      those whole traits?
+        let Some(kyber_prekey_id) =
+            protocol_store.last_resort_kyber_prekey_id().await?
+        else {
+            tracing::warn!("No last resort kyber prekey found");
+            return Ok(false);
+        };
+
+        let signed_prekey =
+            protocol_store.get_signed_pre_key(signed_prekey_id).await?;
+        let kyber_prekey =
+            protocol_store.get_kyber_pre_key(kyber_prekey_id).await?;
```

Comment:
Out of scope, but it bugs me that the function names have `pre_key` instead of `prekey` :sweat_smile:

Reply:
We should indeed get those all in order. And we should change all the occurances of i32 in our own traits to the type safe versions provided by lsc.